### ISSUE TITLE
[Feature proposal] Allow python scikit-learn wrappers to use callbacks

### ISSUE THREAD

**Issue by mrgutkun** [dmlc/xgboost#3663]
Pretty much what it says on the tin. Being able to use xgboost in a scikit pipeline is terribly convenient, and so is being able to finely control learning rate, but at the moment it's one or the other.
As I understand, making wrappers' `fit()` methods pass callbacks to the `train()` inside would do the job, but maybe there are reasons not to do it like this, or not to do it at all?

**Comment by hcho3** [comment]
@mrgutkun Yes, LightGBM does a similar thing, passing a callback to `fit()` method: https://lightgbm.readthedocs.io/en/latest/Python-API.html#lightgbm.LGBMModel.fit . I'll gladly merge a pull request to enable this.
